R/create_individual_file.R

```diff
@@ -0,0 +1,924 @@
+#' Create individual file
+#'
+#' @description Creates individual file from episode file
+#'
+#' @param episode_file Tibble containing episodic data
+create_individual_file <- function(episode_file) {
+  episode_file %>%
+    remove_blank_chi() %>%
+    find_non_duplicates(.data$cij_marker, "Distinct_CIJ") %>%
+    add_cij_columns() %>%
+    find_non_duplicates(.data$ch_chi_cis, "first_ch_ep") %>%
+    add_all_columns() %>%
+    find_non_duplicates(.data$ooh_case_id, "unique_ooh_case") %>%
+    dplyr::mutate(unique_ooh_case = dplyr::if_else(recid != "OoH", 0, unique_ooh_case)) %>%
+    aggregate_ch_episodes() %>%
+    clean_up_ch() %>%
+    recode_gender() %>%
+    aggregate_by_chi() %>%
+    clean_individual_file()
+}
+
+#' Remove blank CHI
+#'
+#' @description Convert blank strings to NA and remove NAs from CHI column
+#'
+#' @inheritParams create_individual_file
+remove_blank_chi <- function(episode_file) {
+  episode_file %>%
+    dplyr::mutate(chi = dplyr::na_if(.data$chi, "")) %>%
+    dplyr::filter(!is.na(.data$chi))
+}
+
+#' Find non-duplicates
+#'
+#' @description Create new column which marks first (per group)
+#' non-duplicated observation as 1, with any duplicates marked as 0.
+#'
+#' @inheritParams create_individual_file
+#' @param group Column to group by
+#' @param col_name Name of new column
+find_non_duplicates <- function(episode_file, group, col_name) {
```

Comment:
This can be refactored and used `n_distinct(cij_marker)` in the aggregate later on in `aggregate_by_chi`